Source/WebCore/platform/graphics/Image.h

```diff
@@ -203,7 +208,10 @@ class Image : public RefCounted<Image> {
 #if PLATFORM(WIN)
     virtual void drawFrameMatchingSourceSize(GraphicsContext&, const FloatRect& dstRect, const IntSize& srcSize, CompositeOperator) { }
 #endif
+    virtual bool canDrawViewport(GraphicsContext&) const { return false; }
+    virtual bool shouldDrawViewport(GraphicsContext&) const { return false; }
     virtual ImageDrawResult draw(GraphicsContext&, const FloatRect& dstRect, const FloatRect& srcRect, const ImagePaintingOptions& = { }) = 0;
+    ImageDrawResult drawViewport(GraphicsContext&, const FloatRect& dstRect, const FloatRect& srcRect, const ImagePaintingOptions& = { });
```

Comment:
Can this be private and ImageViewport made a friend class? Having this public makes things confusing.

Reply:
`drawViewport()` is a protected method of `Image`. And `ImageViewport` is a friend of `Image`.